### PR TITLE
File popup mariana

### PR DESCRIPTION
**Pull request by mariana-novoa**
Hi so in this branch we worked on creating a right-click menu for files and directories. The popup menu includes create Text File, delete File, copy and paste. It shows you a different one whether you select a directory or a file. Additionally, since Dario also had a right click popup menu for the management of files (close all files, etc), we decided to incorporate both into a single class to adhere to Clean Architecture. Note that we are missing some Javadoc comments.

### CODE REVIEW

**Review by dariosanp — APPROVED** [review]
The new code seems to merge with no issues. The conflicts seem to be caused by removing unnecessary code. I Approve of adding your version.

File: src/main/DarioMarianaTest/use_case/EditorOperations/EditorOperationsTest.java
Comment (marburcea) [inline src/main/DarioMarianaTest/use_case/EditorOperations/EditorOperationsTest.java:79]: Great test case and coverage, but make sure to move it to test file before merging!

File: src/main/java/entity/ParentIDEJTabbedPane.java
Comment (marburcea) [inline src/main/java/entity/ParentIDEJTabbedPane.java:125]: Way to move the logic for this code to another file and make it more abstract

File: src/main/java/view/PopupMenuHandler.java
Comment (marburcea) [inline src/main/java/view/PopupMenuHandler.java:168]: improved way to handle a pop up

**Comment by marburcea** [comment]
Approved overall

**Comment by yaraalkurdi** [comment]
Approved

File: src/main/java/entity/Editor.java
Comment (AriGood) [inline src/main/java/entity/Editor.java:1]: by removing this are we just pushing logic downwards

**Review by AriGood — CHANGES_REQUESTED** [review]
.idea folder should not be here please don't merge well it is there

also you should add docstrings and error handling before merging

File: src/main/java/entity/Editor.java
Comment (dariosanp) [reply src/main/java/entity/Editor.java:1]: Editor served no purpose as it was replaced by EditorObj

**Pull request merged**
